src/ripple/ledger/ViewAPI.h

```diff
@@ -208,8 +214,9 @@ trustCreate (View& view,
                                         // Issuer should be noAccount()
     STAmount const& saLimit,            // --> limit for account being set.
                                         // Issuer should be the account being set.
-    const std::uint32_t uSrcQualityIn = 0,
-    const std::uint32_t uSrcQualityOut = 0);
+    std::uint32_t uSrcQualityIn,
+    std::uint32_t uSrcQualityOut,
+    CreditHook hook);
```

Comment:
@seelabs I might put the functions that take the hook up as members in the `BasicView` and `View` classes, and make the hook function a protected overloadable `virtual`. This will hide the hook and also allow customizations.